Modules/NetworkSecurityGroups/2.0/Scripts/enable.flow.logs.ps1

```diff
@@ -0,0 +1,62 @@
+[CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]
+        $SubscriptionId,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $NetworkWatcherRegion,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $NetworkSecurityGroupId,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $DiagnosticStorageAccountId,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $WorkspaceId,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $LogAnalyticsWorkspaceId,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $WorkspaceRegion
+    )
+
+try {
+    $currentSubscriptionId = Get-AzContext | Select-Object "Subscription"
+
+    if ($null -ne $currentSubscriptionId -and `
+        $currentSubscriptionId.Subscription.Id -eq $SubscriptionId) {
+        Set-AzContext -Subscription $SubscriptionId
+    }
+    else {
+        throw "Are you logged in? Please, make sure to run Login-AzAccount"
+    }
+
+    $WorkspaceRegion = $WorkspaceRegion.Replace(' ', '').ToLower()
+    $NetworkWatcherRegion = $NetworkWatcherRegion.Replace(' ', '').ToLower()
+
+    $registered = Get-AzResourceProvider -ProviderNamespace Microsoft.Insights
+
+    if ($null -eq $registered) {
+        Register-AzResourceProvider -ProviderNamespace Microsoft.Insights
+    }
+
+    While ($null -eq $registered) { 
```

Comment:
I believe Register-AzResourceProvider is async. Can the registration fail after Register-AzResourceProvider is accepted but not completed successfully? If so, this might end up being an infinite loop.